unittests/test_policies.py

```diff
@@ -383,3 +393,174 @@ def test_poll_fails_busy_loop(self):
         stats = self.runner.stats
         self.assertEqual(num_tasks, stats.num_cases())
         self.assertEqual(num_tasks, len(stats.failures()))
+
+
+class TestDependencies(unittest.TestCase):
+    class Node:
+        """A node in the test case graph.
+
+        It's simply a wrapper to a (test_name, partition, environment) tuple
+        that can interact seemlessly with a real test case.
+        It's meant for convenience in unit testing.
+        """
+
+        def __init__(self, cname, pname, ename):
+            self.cname, self.pname, self.ename = cname, pname, ename
+
+        def __eq__(self, other):
+            if isinstance(other, type(self)):
+                return (self.cname == other.cname and
+                        self.pname == other.pname and
+                        self.ename == other.ename)
+
+            if isinstance(other, executors.TestCase):
+                return (self.cname == other.check.name and
+                        self.pname == other.partition.fullname and
+                        self.ename == other.environ.name)
+
+            return NotImplemented
+
+        def __hash__(self):
+            return hash(self.cname) ^ hash(self.pname) ^ hash(self.ename)
+
+        def __repr__(self):
+            return 'Node(%r, %r, %r)' % (self.cname, self.pname, self.ename)
+
+    def has_edge(graph, src, dst):
+        return dst in graph[src].deps
+
+    def num_deps(graph, cname):
+        return sum(len(c.deps) for c in graph.values()
+                   if c.check.name == cname)
+
+    def find_check(name, checks):
+        for c in checks:
+            if c.name == name:
+                return c
+
+        return None
+
+    def find_case(cname, ename, cases):
+        for c in cases:
+            if c.check.name == cname and c.environ.name == ename:
+                return c
+
+    def setUp(self):
+        self.loader = RegressionCheckLoader([
+            'unittests/resources/checks_unlisted/dependencies/normal.py'
+        ])
+
+        # Set runtime prefix
+        rt.runtime().resources.prefix = tempfile.mkdtemp(dir='unittests')
+
+    def tearDown(self):
+        os_ext.rmtree(rt.runtime().resources.prefix)
+
+    @rt.switch_runtime(fixtures.TEST_SITE_CONFIG, 'sys0')
+    def test_build_deps(self):
+        Node = TestDependencies.Node
+        has_edge = TestDependencies.has_edge
+        num_deps = TestDependencies.num_deps
+        find_check = TestDependencies.find_check
+        find_case = TestDependencies.find_case
+
+        cases = executors.generate_testcases(self.loader.load_all())
+        deps = dependency.build_deps(cases)
+
+        # Check DEPEND_FULLY dependencies
+        assert num_deps(deps, 'Test1_fully') == 8
```

Comment:
I switched intentionally to pytest notation. I find it less verbose and more integrated to the language. I suggest new unit tests to follow this notation and gradually migrate the older tests.